src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.github.lindenb.jbwa.jni.AlnRgn;
+import com.github.lindenb.jbwa.jni.BwaIndex;
+import com.github.lindenb.jbwa.jni.BwaMem;
+import com.github.lindenb.jbwa.jni.ShortRead;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.bwa.BWANativeLibrary;
+import scala.Tuple2;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ContigAligner implements Closeable {
+
+    final BwaIndex index;
+    final BwaMem bwaMem;
+
+    public ContigAligner(final File referenceFile) {
+        BWANativeLibrary.load();
+        try {
+            index = new BwaIndex(referenceFile);
+        } catch (IOException e) {
+            throw new GATKException("Unable to load reference", e);
+        }
+        bwaMem = new BwaMem(index);
+
+    }
+
+    public List<AssembledBreakpoint> alignContigs (final RunSGAViaProcessBuilderOnSpark.ContigsCollection contigsCollection) {
+        final List<AssembledBreakpoint> assembledBreakpoints = new ArrayList<>();
+
+
+            try {
+                for(final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contigInfo : contigsCollection.getContents()) {
+                    final String contigId = contigInfo._1.toString();
+                    final byte[] sequence = contigInfo._2.toString().getBytes();
+                    final AlnRgn[] alnRgns = bwaAlignSequence(bwaMem, contigId, sequence);
+
+                    // todo: parse cigar for internal indels
+                    if (alnRgns.length > 1) {
+
+                        // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                        final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                                .filter(a -> a.getSecondary() == -1)
+                                .map(AlignmentRegion::new)
+                                .sorted(Comparator.comparing(a -> a.startInContig))
+                                .collect(Collectors.toList());
+
+                        AlignmentRegion current = alignmentRegionList.get(0);
```

Comment:
You don't know that there is a zero'th element.  It's a pretty safe assumption, because `alnRgns.length > 1` and it would be very weird if all the alignments were secondary so that they all got filtered out.  But I'd at least add a comment something like:
`// Assuming that at least one of the alignments is primary and so there will be at least one element remaining.`